runtime-modules/council/src/lib.rs

```diff
@@ -738,7 +790,44 @@ impl<T: Trait> EnsureChecks<T> {
         }
 
         // ensure that previous stake is unlockable (if any present)
-        Self::ensure_previous_stake_unlockable(council_user_id, &account_id)?;
+        Self::ensure_previous_stake_reusable(council_user_id, &account_id)?;
+
+        Ok(account_id)
+    }
+
+    /// Ensures there is no problem in withdrawing already announced candidacy.
+    fn can_withdraw_candidacy(
+        origin: T::Origin,
+        council_user_id: &T::CouncilUserId,
+    ) -> Result<T::AccountId, Error<T>> {
+        // ensure user's membership
+        let account_id = Self::ensure_user_membership(origin, council_user_id)?;
+
+        // escape when no previous candidacy stake is present
+        if !Candidates::<T>::contains_key(council_user_id) {
+            return Err(Error::NotCandidatingNow);
+        }
+
+        let candidate = Candidates::<T>::get(council_user_id);
+
+        // ensure candidacy was announced in on of previous election cycles
+        if candidate.cycle_id != CurrentAnnouncementCycleId::get() {
+            return Err(Error::NotCandidatingNow);
+        }
+
+        // ensure candidacy announcing period is running now
+        match Stage::<T>::get().stage {
+            CouncilStage::Announcing(_) => (),
+            _ => return Err(Error::CantWithdrawCandidacyNow),
+        };
+
+        // ensure there is no mismatch between saved and current `council_user_id`/`account_id` combination
+        if candidate.account_id != account_id {
```

Comment:
`account_id` is the controller account of the member, but `Candidate::account_id` is the staking account of the candidate, these are not the same in general. Perhaps renaming `account_id` to `staking_account_id` would increase clarity to avoid such possible confusion.

Reply:
`staking_account_id` name was introduced and withdrawal no longer needs to compare those ids (you can authorize withdrawal with different `account_id` as long as you are using the same membership id).